js/views/popover.jsx

```diff
@@ -29,27 +29,30 @@ class Popover extends React.Component {
    * @param {event} event description
    */
   onKeyDown = (event) => {
-    if (!this.props.closeActionEnabled || typeof this.props.closeAction !== 'function') {
+    const { closeActionEnabled, closeAction } = this.props;
+    if (!closeActionEnabled || typeof closeAction !== 'function') {
       return;
     }
     if (event.which === CONSTANTS.KEYCODES.ESCAPE_KEY || event.keyCode === CONSTANTS.KEYCODES.ESCAPE_KEY) {
       // Ask parent to restore the focus of the toggle button that triggers this
       // popover if the ESC key was pressed while the focus was inside this element.
       // If the focus was outside the popover we shouldn't re-focus the toggle button.
       const targetIsChildElement = this.domElement ? this.domElement.contains(event.target) : false; // eslint-disable-line
-      this.props.closeAction({
+      closeAction({
         restoreToggleButtonFocus: targetIsChildElement,
       });
     }
   }
 
   render() {
+    const { children, dir, popoverClassName } = this.props;
     return (
       <div
+        dir={dir}
```

Comment:
do you still need `dir` attribute for divs?

Reply:
+